privilege/privileges/privileges.go

```diff
@@ -462,11 +553,11 @@ func (p *UserPrivileges) ConnectionVerification(user *auth.UserIdentity, authUse
 			hpwd, err := auth.DecodePassword(pwd)
 			if err != nil {
 				logutil.BgLogger().Error("decode password string failed", zap.Error(err))
-				return ErrAccessDenied.FastGenByArgs(user.Username, user.Hostname, hasPassword)
+				return &ErrUserPasswordFailed{ErrAccessDenied.FastGenByArgs(user.Username, user.Hostname, hasPassword)}
```

Comment:
Please add the following comment in the next line of `switch record.AuthPlugin {`
```go
// NOTE: If the checking of the password fails, please return the error wrap by ErrUserPasswordFailed.
```

Reply:
done